File: week15/week15-5.cpp
//week15-5.c
int equalSubstring(char* s, char* t, int maxCost) {
    int ans=0,j=0;
    int N=strlen(s);//先知道字串長度
    for(int i=0;i<N;i++){//字串的迴圈
        maxCost-=abs(s[i]-t[i]);
        while (maxCost<0){
            maxCost+=abs(s[j]-t[j]);
            j+=1;
        }
        if(i-j+1>ans) ans=i-j+1;
    }
    return ans;//作業-1
}

File: week15/week15-5.py
class Solution: #week15-5.py
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        #先用python把字串迴圈寫出來
        ans=0
        N=len(s)
        j=0#尾巴
        for i in range(N):#頭是i 先用python把字串的迴圈寫出來
            maxCost-=abs(ord(s[i])-ord(t[i]))#毛毛蟲的頭i吃葉子
            while maxCost<0:#預算不足，變負的
                maxCost+=abs(ord(s[j])-ord(t[j]))#尾巴j大出來
                j+=1 #尾巴j往右縮
            ans=max(ans,i-j+1) #頭i-尾j+1就是毛毛蟲的長度
        return ans